### PR TITLE
Update models

### LINKED ISSUE #96: Update models

### PR DESCRIPTION
**Pull request by ganhongyao**
- Add charity fields
  - name
  - description
  - websiteUrl
- Add charity attachments
  - logo
  - image
- Add Giving SG url to campaign charity
- Add validations
  - Campaign promised amount and denomination cannot be changed once persisted
  - Donation's coupon must be meant for that campaign
  - Donation's coupon cannot be a redeemed one on creation
  - Donation's coupon cannot be changed once persisted

Closes #96

### CODE REVIEW

File: backend/app/models/campaign.rb [hunk 1]
```diff
@@ -17,6 +19,8 @@ class Campaign < ApplicationRecord
   validates :start, presence: true
   validates :end, comparison: { greater_than: :start }
   validates :charities, presence: true
+  validates :promised_amount, final: true
+  validates :coupon_denomination, final: true
```
Comment (sivayogasubramanian) [inline backend/app/models/campaign.rb:23]: I am not testing this. Will trust that this works.

File: backend/app/models/secondary_donation.rb [hunk 2]
```diff
@@ -5,4 +5,26 @@ class SecondaryDonation < ApplicationRecord
...
+
+    coupon_campaign = coupon.campaign
+    campaign_charity_campaign = campaign_charity.campaign if campaign_charity
+
+    return if coupon_campaign == campaign_charity_campaign
+
+    errors.add(:coupon, 'must be for the same campaign as the campaign charity')
+  end
+
+  def coupon_unredeemed
```
Comment (sivayogasubramanian) [inline backend/app/models/secondary_donation.rb:25]: Do we need to check for coupon.nil?

**Review by sivayogasubramanian — APPROVED** [review]
LGTM!

File: backend/app/models/secondary_donation.rb (hunk 2 above)
Comment (ganhongyao) [reply backend/app/models/secondary_donation.rb:25]: Yup, thanks for the spot

**Pull request merged**